src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136355,6 +136355,193 @@ static void validate(ModuleSymbol module)
+                }
+
+                interface I<T>
+                {
+                    {{def}}
+                }
+                """;
+            var verifier = CompileAndVerify(source, targetFramework: TargetFramework.Net60,
+                verify: ExecutionConditionUtil.IsDesktop ? Verification.Skipped : Verification.Passes,
+                symbolValidator: validate);
+            verifier.VerifyDiagnostics(
+                // (2,7): warning CS8645: 'I<T>' is already listed in the interface list on type 'C<T>' with different nullability of reference types.
+                // class C<T> : I<T?>, I<T>
+                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "C").WithArguments("I<T>", "C<T>").WithLocation(2, 7));
+
+            static void validate(ModuleSymbol module)
+            {
+                var cm = module.GlobalNamespace.GetMember<MethodSymbol>("C.M");
+                var cme = Assert.Single(cm.GetExplicitInterfaceImplementations());
+                var im = module.GlobalNamespace.GetMember<MethodSymbol>("I.M");
+                Assert.Same(im, cme.OriginalDefinition);
+            }
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/73338")]
+        [InlineData("static abstract void M();")]
+        [InlineData("static virtual void M() { }")]
+        public void AnnotationMismatch_Interface_StaticVirtualMethod_04(string def)
+        {
+            var source = $$"""
+                #nullable enable
+                class C<T> : I<T?>, I<T>
+                {
+                    static void I<T>.M() { }
+                }
+
+                interface I<T>
+                {
+                    {{def}}
+                }
+                """;
+            var verifier = CompileAndVerify(source, targetFramework: TargetFramework.Net60,
+                verify: ExecutionConditionUtil.IsDesktop ? Verification.Skipped : Verification.Passes,
+                symbolValidator: validate);
+            verifier.VerifyDiagnostics(
+                // (2,7): warning CS8645: 'I<T>' is already listed in the interface list on type 'C<T>' with different nullability of reference types.
+                // class C<T> : I<T?>, I<T>
+                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "C").WithArguments("I<T>", "C<T>").WithLocation(2, 7));
+
+            static void validate(ModuleSymbol module)
+            {
+                var cm = module.GlobalNamespace.GetMember<MethodSymbol>("C.I<T>.M");
+                var cme = Assert.Single(cm.GetExplicitInterfaceImplementations());
+                var im = module.GlobalNamespace.GetMember<MethodSymbol>("I.M");
+                Assert.Same(im, cme.OriginalDefinition);
+            }
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/73338")]
+        public void AnnotationMismatch_Interface_StaticVirtualMethod_05(
```

Comment:
>AnnotationMismatch_Interface_StaticVirtualMethod_05 [](http://example.com/codeflow?start=20&length=51)

I assume this problem can be reproduced with instance methods as well. Let's add a test for that scenario too.